blocks_vertical/data.js

```diff
@@ -206,6 +206,66 @@ Blockly.Blocks['data_listcontents'] = {
   }
 };
 
+Blockly.Blocks['data_listindexall'] = {
+  /**
+   * List index menu, with all option.
+   * @this Blockly.Block
+   */
+  init: function() {
+    this.jsonInit({
+      "message0": "%1",
+      "args0": [
+        {
+          "type": "field_numberdropdown",
+          "name": "INDEX",
+          "value": "1",
+          "min": 0,
+          "precision": 1,
+          "options": [
+            ["1", "1"],
+            ["last", "last"],
+            ["all", "all"]
+          ]
+        }
+      ],
+      "output": "String",
+      "category": Blockly.Categories.data,
+      "outputShape": Blockly.OUTPUT_SHAPE_ROUND,
+      "colour": Blockly.Colours.textField
+    });
+  }
+};
+
+Blockly.Blocks['data_listindexrandom'] = {
```

Comment:
Hmmm, they have different drop-down menus (one contains "all" as an option (for delete), one contains "random"). Not sure how to do this otherwise?